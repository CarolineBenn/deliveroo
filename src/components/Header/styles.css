@media (min-width: 678px) { 
  .header-inner-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
  }
}

.header {
  background-color: #000000; 
  height: 45px;
  width: 100%;
  color: white;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  z-index: 1030;
}

@media (min-width: 678px) {
  .header {
    background-color: transparent;
    position: absolute;
    top: 0;
    left: 0;
    height: 5rem;
    flex-direction: row;
    text-transform: uppercase;
    font-size: 14px;
  }
}

.site-name {
  text-transform: uppercase;
  text-decoration: none;
}

.site-name:visited {
  color: #FFFFFF;
}

.menu-icon {
  height: 1.5rem;
  width: 1.5rem;
  position: absolute;
  right: 1rem;
  top: calc(50% - 0.15rem);
  transform: translateY(-50%);
  outline: none;
}

@media (min-width: 678px) {
  .menu-icon {
    display: none;
  }
}

.bar {
  display: block;
  width: 100%;
  height: 0.20rem;
  margin: 0.30rem auto;
  border-radius: 5px;
  background: #FFFFFF;
}

.logo {
  max-width: 200px;
}