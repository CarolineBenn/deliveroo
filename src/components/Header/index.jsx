import React, { Component } from 'react';

import HeaderNav from './components/HeaderNav';

import './styles.css';

class Header extends Component {
  constructor() {
    super();
    this.state = {
      showNav: false,
    };
  }

  toggleNav = () => {
    this.setState({ showNav: !this.state.showNav })
  }

  render() {
    const { showNav } = this.state;
    return (
      <header className="header">
        <div className="container header-inner-container">
          <a className="site-name" href="/">
            <img
            src="/images/logo.png"
            className="logo hideBelowMD"
            alt="Deliveroo Logo"
            title="Deliveroo Logo"
          />
            <span className="srOnly hideBelowMD">Deliveroo</span>
            <span className="hideAboveMD brand-name">Deliveroo</span>
          </a>
          <button className="reset menu-icon" onClick={this.toggleNav}>
            <span className="srOnly">Toggle navigation</span>
            <span className="bar"></span>
            <span className="bar"></span>
            <span className="bar"></span>
          </button>
          {showNav &&
            // `mobile` prop indicates that HeaderNav is for mobile size only.
            <HeaderNav key="SM" mobile />}
          <HeaderNav key="MDLG" />
        </div>
      </header>
    );
  }
}

export default Header;
