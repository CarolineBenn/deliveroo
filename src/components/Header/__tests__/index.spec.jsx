import React from 'react';
import renderer from 'react-test-renderer';

import Header from '..';

// jest.mock('../components/HeaderNav', () => 'HeaderNav');

test('Header renders correctly', () => {
  const tree = renderer
    .create(<Header navId="menuNav" />).toJSON();
  expect(tree).toMatchSnapshot();
});
