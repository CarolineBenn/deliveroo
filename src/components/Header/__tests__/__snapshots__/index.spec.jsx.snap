// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="container header-inner-container"
  >
    <a
      className="site-name"
      href="/"
    >
      <img
        alt="Deliveroo Logo"
        className="logo hideBelowMD"
        src="/images/logo.png"
        title="Deliveroo Logo"
      />
      <span
        className="srOnly hideBelowMD"
      >
        Deliveroo
      </span>
      <span
        className="hideAboveMD brand-name"
      >
        Deliveroo
      </span>
    </a>
    <button
      aria-controls="menuNav"
      className="reset menu-icon"
      onClick={[Function]}
    >
      <span
        className="srOnly"
      >
        Toggle navigation
      </span>
      <span
        className="bar"
      />
      <span
        className="bar"
      />
      <span
        className="bar"
      />
    </button>
    <nav
      aria-expanded={false}
      className="header-nav"
      id="menuNav"
    >
      <ul
        className="unstyledList header-nav-menu"
      >
        <li>
          <a
            className="header-nav-menu-item"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="header-nav-menu-item"
            href="/"
          >
            Search
          </a>
        </li>
        <li>
          <a
            className="header-nav-menu-item"
            href="/"
          >
            Â£8.55
          </a>
        </li>
        <li>
          <a
            className="header-nav-menu-item"
            href="/"
          >
            Simon Rohrbach
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
